File: __prologue_server/map/tile.go
/* */

package mymap

type Tile struct {
	Terrain  string `json:"terrain"`
	Walkable bool   `json:"walkable"`
	BlockLOS bool   `json:"blockLOS"`
	Explored bool   `json:"explored"`
	Visible  bool   `json:"visible"`
}

func (t Tile) create(terrain string) Tile {
	switch terrain {
	case "floor":
		return Tile{terrain, true, false, false, false}
	case "wall":
		return Tile{terrain, false, true, false, false}
	case "unknown":
		return Tile{terrain, false, true, false, false}
	}
	return Tile{}
}

File: __prologue_client/core/mapa/tile.js
/* */

console.log('Loading..... core/mapa/tile.js');

class Tile {
  constructor(terrain) {
    switch (terrain) {
      case "floor":
        this.terrain = terrain;
        this.walkable = true;
        this.blockLOS = false;
        break;
      case "wall":
      case "unknown":
        this.terrain = terrain;
        this.walkable = false;
        this.blockLOS = true;
        break;
      default:
        this.terrain = "unknown";
        this.walkable = false;
        this.blockLOS = true;
    }
    this.explored = false;
    this.visible = false;
  }
}

export {
  Tile
};
